### DISCUSSION TITLE
development docs

### DISCUSSION THREAD

**Discussion by leet-code-master** [perspective-dev/perspective#2082] [Q&A] [answered]
In addition to this URL（ https://perspective.finos.org/docs/js/ ）Are there any relevant JavaScript development documents?

**Comment by sinistersnare** [comment, marked as answer]
The [examples](https://perspective.finos.org/examples/) page includes code for a lot of them. A good one is the [SF evictions example](https://perspective.finos.org/block/?example=evictions).

**Reply by leet-code-master** [reply]
I get it.Is there a callback method when exporting or modifying the list sorting?I want to save a configuration state of the current chart on the server side

**Reply by sinistersnare** [reply]
You can use the `workspace.save` and `workspace.restore` methods.

**Reply by sinistersnare** [reply]
Sorry, use `viewer` methods instead. [`viewer.save`](https://perspective.finos.org/docs/obj/perspective-viewer/#save) and [`viewer.restore`](https://perspective.finos.org/docs/obj/perspective-viewer/#restore)

**Reply by leet-code-master** [reply]
For example, if the chart type is modified, [plugin: Y Area]=>[plugin: Y Bar], is there any listening method (similar to Vue's watch method) that can notify the developer that the chart configuration has changed
![image](https://user-images.githubusercontent.com/110711802/215723170-327c7394-dd31-4391-a692-02dad415ebdf.png)

**Comment by texodus** [comment]
Listen to the `perspective-config-update` event [as covered in the docs section "Update Events"](https://perspective.finos.org/docs/js/#update-events):

```js
viewer.addEventListener(event => {
    console.log(event.detail.config.plugin);
});
```
